src/adapter/redstone/RedstoneCoreOracle.sol

```diff
@@ -76,67 +67,47 @@ contract RedstoneCoreOracle is PrimaryProdDataServiceConsumerBase, BaseAdapter {
-        asView(_validateTimestamp)(timestampMillis);
-    }
-
-    /// @notice Validate the timestamp of a Redstone signed price data package and cache the response.
-    /// @param timestampMillis Data package timestamp in milliseconds.
-    /// @dev The price timestamp must lie in the defined acceptance range relative to `block.timestamp`.
+    /// @notice Ingest a Redstone payload, decode and verify it, and cache the price in storage.
+    /// @param timestamp The expected timestamp of the Redstone payload. All data packages must have this timestamp.
+    /// @dev Redstone payload must be appended at the end of the abi-encoded calldata to this function.
+    /// Decoding and validation inherited from `PrimaryProdDataServiceConsumerBase`.
+    /// The price timestamp must lie in the defined acceptance range relative to `block.timestamp`.
     /// Note: The Redstone SDK allows the price timestamp to be up to 1 minute in the future.
-    function _validateTimestamp(uint256 timestampMillis) internal {
-        // The `updatePriceContext` guard effectively blocks external / direct calls to `validateTimestamp`.
+    function updatePrice(uint48 timestamp) external {
+        // The cache can only be updated if it has a more recent timestamp.
         Cache memory _cache = cache;
-        if (_cache.updatePriceContext != FLAG_UPDATE_PRICE_ENTERED) revert Errors.PriceOracle_InvalidAnswer();
-
-        uint256 timestamp = timestampMillis / 1000;
-        // Avoid redundant storage writes as `validateTimestamp` is called for every signer in the payload (3 times).
-        // The inherited Redstone consumer contract enforces that the timestamps are the same for all signers.
-        if (timestamp == _cache.priceTimestamp) return;
+        if (timestamp <= _cache.priceTimestamp) return; // Do not revert to avoid DoS attacks.
 
         if (block.timestamp > timestamp) {
             // Verify that the timestamp is not too stale.
-            uint256 priceStaleness = block.timestamp - timestamp;
-            if (priceStaleness > maxStaleness) {
-                revert Errors.PriceOracle_TooStale(priceStaleness, maxStaleness);
+            uint256 staleness = block.timestamp - timestamp;
+            if (staleness > maxStaleness) {
+                revert Errors.PriceOracle_TooStale(staleness, maxStaleness);
             }
         } else if (timestamp - block.timestamp > RedstoneDefaultsLib.DEFAULT_MAX_DATA_TIMESTAMP_AHEAD_SECONDS) {
             // Verify that the timestamp is not too long in the future (1 min). Redstone SDK explicitly allows this.
             revert Errors.PriceOracle_InvalidAnswer();
         }
 
-        // Enforce that cached price updates have a monotonically increasing timestamp.
-        if (timestamp < _cache.priceTimestamp) revert Errors.PriceOracle_InvalidAnswer();
-        cache.priceTimestamp = uint48(timestamp);
+        // Optimistically update the price timestamp.
+        cache.priceTimestamp = timestamp;
```

Comment:
wouldn't we save one SLOAD doing it this way?
`cache = Cache({price: _cache.price, priceTimestamp: timestamp});`

Reply:
Yep, saves 20-30 median gas.